src/game/server/baseentity.cpp

```diff
@@ -1770,6 +1770,10 @@ int CBaseEntity::TakeDamage( const CTakeDamageInfo &inputInfo )
 float CBaseEntity::GetAttackDamageScale( CBaseEntity *pVictim )
 {
 	float flScale = 1;
+    
+	if ( m_DamageModifiers.Count() <= 0 )
+		return flScale;
+    
```

Comment:
Rather than an empty check, may be worth doing a null check here for m_DamageModifiers; I don't think the GetAttacker() change actually does anything, and I suspect your null pointer crash is coming from damagemodifiers being fully null instead. Same for the other case.